### ISSUE TITLE
[23.1] Investigate image size increases from 23.0 to 23.1 for quarkus-full-microprofile

### ISSUE THREAD

**Issue by jerboaa** [Karm/mandrel-integration-tests#217]
The `23.1` release triggered some size increase checks. Here is one failure with quarkus `3.5.0.CR1`:

```
org.opentest4j.AssertionFailedError: 

Application QUARKUS_FULL_MICROPROFILE in mode none executable size  is 76624 kB, which is over 73500 kB threshold by 4%.
 ==> expected: <true> but was: <false>
	at org.graalvm.tests.integration.RuntimesSmokeTest.testRuntime(RuntimesSmokeTest.java:162)
	at org.graalvm.tests.integration.RuntimesSmokeTest.quarkusFullMicroProfile(RuntimesSmokeTest.java:192)
```

The comparison between `23.0` and `23.1` isn't an apples to apples comparison as `23.0` is JDK 17 based. `23.1` is JDK 21 based. Either way, we need to investigate in order to ensure this isn't unintentional.

**Comment by jerboaa** [comment]
@zakkak You've looked into this recently for 23.0, so assigning to you for now.

**Comment by zakkak** [comment]
The analysis is complete and documented in https://github.com/quarkusio/quarkusio.github.io/pull/1827

TLDR:

According to our analysis the binary size increase is attributed to two distinct changes, both of which are necessary for getting more accurate profiles when using the [async-profiler](https://github.com/async-profiler/async-profiler).

- [**Add support for profiling of topmost frame**](https://github.com/oracle/graal/pull/7003)
- [**ProfilingSampler does not need local variable values**](https://github.com/oracle/graal/pull/6763) (specifically [the commit "Always store bci in frame info"](https://github.com/oracle/graal/commit/d747c30c7691012c39989a8597fd850c68b740ad))

If you think that this kind of info should only be included when the user opts-in, please provide your feedback in [this discussion](https://github.com/oracle/graal/discussions/7707).
